src/com/gfredericks/debug_repl/async.clj

```diff
@@ -0,0 +1,59 @@
+(ns com.gfredericks.debug-repl.async
+  (:require
+    [com.gfredericks.debug-repl :as debug-repl]
+    [com.gfredericks.debug-repl.util :as util])
+  (:import (java.util.concurrent
+            ArrayBlockingQueue
+            TimeUnit)))
+
+(def the-executor nil) ;; global mutable state
+
+(defn can-break?
+  []
+  ;; this just checks if we're in a repl
+  (boolean
+    (if (util/require? 'nrepl.server)
+      @(resolve 'nrepl.middleware.interruptible-eval/*msg*)
+      @(resolve 'clojure.tools.nrepl.middleware.interruptible-eval/*msg*))))
+
+(defmacro break!
+  [& args]
+  `(if (can-break?)
+     (debug-repl/break! ~@args)
+     (if-let [e# the-executor]
+       (e# #(debug-repl/break! ~@args))
+       :noop)))
+
+(defn wait-for-breaks
+  []
+  ;; if the repl is busy executing a request, should concurrent
+  ;; requests block or skip the repl execution altogether?
+  ;;
+  ;; going to try blocking first
```

Comment:
this comment is pretty important -- probably best to make this function able to do both, based on an arg

Reply:
Doesn't this need to be controlled from `break!`, not from `wait-for-breaks`? Or am I misunderstanding the meaning here?

Reply:
I think you could do it either way, but I'm not sure how you would do it exactly; I think if you want to make it not-blocking, replacing the ArrayBlockingQueue with a SynchronousQueue would let you detect whether the `loop` can execute something; that might constrain it to be implemented in `wait-for-breaks`